src/Stratis.Bitcoin/Consensus/ConsensusRuleEngine.cs

```diff
@@ -220,6 +226,13 @@ private async Task ExecuteRulesAsync(IEnumerable<AsyncConsensusRule> asyncRules,
             catch (ConsensusErrorException ex)
             {
                 ruleContext.ValidationContext.Error = ex.ConsensusError;
+
+                uint256 hashToBan = ruleContext.ValidationContext.ChainedHeaderToValidate.HashBlock;
+
+                if (ruleContext.ValidationContext.RejectUntil != null)
```

Comment:
Here should be 2nd rule that doesn't lead to permanent block hash banning. Who remembers what is it?

Reply:
Fixed. It was mutated merkle tree exception